### PR TITLE
XRDDEV-1319 Permission checks

### PR DESCRIPTION
**Pull request by Riippi**
Jira ticket: https://jira.niis.org/browse/XRDDEV-1319

### CODE REVIEW

**Review by TJaakkola — APPROVED** [review]
Just a typo in comments

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue [hunk 1]
```diff
@@ -65,29 +63,41 @@ export default Vue.extend({
...
+  computed: {
+    showGenerateCsr(): boolean {
+      // Check if the user has permission to see generate csr action
+      if (this.tokenKey.usage === 'AUTHENTICATION') {
+        return this.$store.getters.hasPermission(
+          Permissions.GENERATE_AUTH_CERT_REQ,
+        );
       }
+      // If key usage is not auth then it has to be sign
+      return this.$store.getters.hasPermission(
```
Comment (jansu76) [inline src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue:79]: Does this work OK if key does not yet have a defined usage (no CSRs have been created) and user has only `generate_auth_cert_req` permission, not `generate_sign_cert_req`? (there is no such role at the moment, though, which would hide the issue)

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue [hunk 2]
```diff
@@ -206,37 +204,37 @@ export default Vue.extend({
...
-      return false;
+
+    canCreateAuthCsr(): boolean {
+      return this.$store.getters.hasPermission(
+        Permissions.GENERATE_AUTH_CERT_REQ,
+      );
+    },
+    canCreateSignCsr(): boolean {
+      return this.$store.getters.hasPermission(
+        Permissions.GENERATE_AUTH_CERT_REQ,
```
Comment (jansu76) [inline src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue:226]: Should be generate_sign_cert_req?

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/UnknownKeysTable.vue [hunk 3]
```diff
@@ -94,9 +94,9 @@ export default Vue.extend({
   },
   computed: {
     hasPermission(): boolean {
```
Comment (jansu76) [inline src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/UnknownKeysTable.vue:96]: Would method name `canCreateCsr` be more descriptive (and same style seems to be used elsewhere)?

**Review by jansu76 — CHANGES_REQUESTED** [review]

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue (hunk 1 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue:79]: If the key doesn't have a usage type it ends up in UnknownKeysTable.vue. That doesn't use the KeyRow component.

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue [hunk 4]
```diff
@@ -206,37 +204,37 @@ export default Vue.extend({
     };
   },
   computed: {
-    hasPermission(): boolean {
+    canImportFromToken(): boolean {
+      // Can the user import cwrtificate from hardware token
```
[the remark this answers is not in the document]
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue:208]: Fixed

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue (hunk 2 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue:226]: Seems that these two methods are not used at all. This is checked in KeyRow component.

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/UnknownKeysTable.vue (hunk 3 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/UnknownKeysTable.vue:96]: Fixed

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue (hunk 2 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTable.vue:226]: I will remove these.

File: src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue (hunk 1 above)
Comment (jansu76) [reply src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue:79]: ah, gotcha :+1:

**Review by jansu76 — APPROVED** [review]

**Pull request merged**